src/System.ComponentModel.Composition/tests/System/ComponentModel/IO/FileIO.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using System.UnitTesting;
+
+namespace System.IO 
+{
+    public static class FileIO
+    {
+        public const string RootTemporaryDirectoryName = "RootTempDirectory";
+
+        private static string _temporaryDirectory;
+        public static string GetRootTemporaryDirectory()
```

Comment:
You'll have to go ahead and replace this file since it's trying to write to the test directory.

Change classes that need this to derive from FileCleanupTestBase and then have them use TestDirectory, GetTestFilePath(), GetTestFileName() instead of the methods in this file